plugins/gateway/kong/plugin.go

```diff
@@ -0,0 +1,179 @@
+	//}
+	destPort := routedService.Port
+	host := routedService.Host
+
+	// Will get the actual path that the request was sent to, not the routed one
+	path, err := kong.Request.GetPath()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request path. %v", err)
+	}
+	reqBody, err := kong.Request.GetRawBody()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request body. %v", err)
+	}
+	resBody, err := kong.ServiceResponse.GetRawBody()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response body. %v", err)
+	}
+	method, err := kong.Request.GetMethod()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request method. %v", err)
+	}
+
+	statusCode, err := kong.ServiceResponse.GetStatus()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response status code. %v", err)
+	}
+	scheme, err := kong.Request.GetScheme()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reuqest scheme. %v", err)
+	}
+	version, err := kong.Request.GetHttpVersion()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request http version. %v", err)
+	}
+	reqHeaders, err := kong.Request.GetHeaders(-1) // default limit of 100 headers
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request headers. %v", err)
+	}
+	// Unlike kong.Response.GetHeaders(), this function will only return headers
+	// that were present in the response from the Service (ignoring headers added
+	// by Kong itself)
+	resHeaders, err := kong.ServiceResponse.GetHeaders(-1) // default limit of 100 headers
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response headers. %v", err)
+	}
+
+	telemetry := models.Telemetry{
+		DestinationAddress:   ":" + strconv.Itoa(destPort), // No destination ip for now
+		DestinationNamespace: "",
+		Request: &models.Request{
+			Common: &models.Common{
+				TruncatedBody: false,
+				Body:          strfmt.Base64(reqBody),
+				Headers:       createHeaders(reqHeaders),
+				Version:       fmt.Sprintf("%f", version),
+			},
+			Host:   parseHost(host),
+			Method: method,
+			Path:   path,
+		},
+		RequestID: "", // TODO from where? do we need?
+		Response: &models.Response{
+			Common: &models.Common{
+				TruncatedBody: false,
+				Body:          strfmt.Base64(resBody),
+				Headers:       createHeaders(resHeaders),
+				Version:       fmt.Sprintf("%f", version),
+			},
+			StatusCode: strconv.Itoa(statusCode),
+		},
+		Scheme:        scheme,
+		SourceAddress: clientIP + ":80",
+	}
+
+	return &telemetry, nil
+}
+
+func parseHost(kongHost string) string {
+	sp := strings.Split(kongHost, ".")
+
+	if len(sp) < 2 {
+		return kongHost
+	}
+
+	return sp[0] + "." + sp[1]
+}
+
+func createHeaders(headers map[string][]string) []*models.Header {
+	ret := []*models.Header{}
+
+	// TODO support multiple values for a header
+	for header, values := range headers {
+		ret = append(ret, &models.Header{
+			Key:   header,
+			Value: values[0],
+		})
+	}
+	return ret
+}
+
+var Version = "0.2"
+var Priority = 1
```

Comment:
what are those?
if that is the plugin version we should set it during build with build flags like we do for the backend

Reply:
The priority is meaningless since we are the only plugin server
The version is also just a thing you need to pass to the server. Not sure where it is being used

Reply:
> since we are the only plugin server
maybe the user has its own plugins

> Not sure where it is being used
Still I would put here the tag/commit hash from the build